src/emulator/commandUtils.ts

```diff
@@ -156,13 +169,59 @@ export function parseInspectionPort(options: any): number {
   return parsed;
 }
 
-export function shutdownWhenKilled(): Promise<void> {
+/**
+ * Sets the correct export options based on --import and --export-on-exit.
+ * Also validates if we have a correct setting we need to export the data on exit.
+ * When used as: `--import ./data --export-on-exit` or `--import ./data --export-on-exit ./data`
+ * we do allow an non-existing --import [dir] and we just export-on-exit. This because else one would always need to
+ * export data the first time they start developing on a clean project.
+ * @param options
+ */
+export function setExportOnExitOptions(options: any): any {
+  if (options.exportOnExit) {
+    if (typeof options.import === "string") {
+      options.exportOnExit =
+        typeof options.exportOnExit === "string" ? options.exportOnExit : options.import;
+
+      const importPath = path.resolve(options.import);
+      if (!fsutils.dirExistsSync(importPath) && options.import === options.exportOnExit) {
+        // --import path does not exist and is tha same as --export-on-exit, let's not import and only --export-on-exit
+        options.exportOnExit = options.import;
+        delete options.import;
+      }
+    }
+
+    if (typeof options.exportOnExit === "string") {
+      return options;
+    }
+    throw new FirebaseError(EXPORT_ON_EXIT_USAGE_ERROR);
+  }
+  return options;
+}
+
+export function shutdownWhenKilled(options: any): Promise<void> {
   return new Promise((res, rej) => {
-    process.on("SIGINT", () => {
-      controller
-        .cleanShutdown()
-        .then(res)
-        .catch(rej);
+    let receivedSIGINTS = 0;
+    process.on("SIGINT", async () => {
+      try {
+        receivedSIGINTS = receivedSIGINTS + 1;
```

Comment:
If I send SIGINT twice won't this logic run `await onExit(options)` and `await controller.cleanShutdown()` more than once?

Reply:
![image](https://user-images.githubusercontent.com/1010260/86251356-0f179d00-bbb2-11ea-95ec-4c9670b6af1b.png)

`onExit` will run only once, `cleanShutdown()` twice and then gives up. Current behaviour is that `cleanShutdown()`  is always called every time you do SIGINT. Not ok when cleanShutdown is hanging on some stop and never finishes or take ages. You need to be able to bail out there.

I was considering changing it to `process.once("SIGINT", ...` and only do everything once but then I was thinking of the case that an onExit takes a long time and the user want to just quit, in that case we still want to do the `cleanShutdown()`. 

Let me know what you think.

Reply:
Ok so after fighting my Windows PC for a few hours I finally got everything running.  The new SIGINT handling seems fine however:

1. When I do a single Ctrl+C I now get warnings like `Firestore Emulator has exited upon receiving signal: SIGINT` which make it seem like something went wrong but I think that's normal?
2. The log message `Received SIGINT 1` is confusing because it feels like 1 is the signal code but it's the count.  Suggest `Received SIGINT (count=1)`

Reply:
Hmmmm actually while playing with this I quickly got into a state where the Database emulator is running free and holding port 9000, so I have to shut it down with `pkill` or similar.  That's not good.